bach/bach/dataframe.py

```diff
@@ -1545,25 +1543,65 @@ def groupby(
         from bach.partitioning import GroupBy, GroupingList, GroupingSet
 
         df = self
-        if self._group_by:
+        if df._group_by:
             # We need to materialize this node first, we can't stack aggregations (yet)
-            df = self.materialize(node_name='nested_groupby')
+            df = df.materialize(node_name='nested_groupby')
 
         group_by: GroupBy
         if isinstance(by, tuple):
+            if not is_postgres(self.engine):
+                raise DatabaseNotSupportedException(
+                    self.engine,
+                    f'GroupingSets are not supported for this SQL dialect. Only grouping by one or more '
+                    f'Series is supported. To use this functionality: specify a Series, a list of Series, a '
+                    f'Series name, or a list of Series names. Dialect: {self.engine.name}')
             # by is a list containing at least one other list. We're creating a grouping set
             # aka "Yo dawg, I heard you like GroupBys, ..."
             group_by = GroupingSet(
                 [GroupBy(group_by_columns=df._partition_by_series(b)) for b in by]
             )
-        elif isinstance(by, list) and len([b for b in by if isinstance(b, (tuple, list))]) > 0:
+            return DataFrame._groupby_to_frame(df, group_by)
+
+        if isinstance(by, list) and len([b for b in by if isinstance(b, (tuple, list))]) > 0:
+            if not is_postgres(self.engine):
+                raise DatabaseNotSupportedException(
+                    self.engine,
+                    f'GroupingLists are not supported for this SQL dialect. Only grouping by one or more '
+                    f'Series is supported. To use this functionality: specify a Series, a list of Series, a '
+                    f'Series name, or a list of Series names. Dialect: {self.engine.name}')
             group_by = GroupingList(
                 [GroupBy(group_by_columns=df._partition_by_series(b)) for b in by])
-        else:
-            by_mypy = cast(Union[str, 'Series',
-                                 List[DataFrame._GroupBySingleType], None], by)
-            group_by = GroupBy(group_by_columns=df._partition_by_series(by_mypy))
+            return DataFrame._groupby_to_frame(df, group_by)
 
+        # This is the normal group-by case
+        by_mypy = cast(Union[str, 'Series',
+                             List[DataFrame._GroupBySingleType], None], by)
+        group_by_columns = df._partition_by_series(by_mypy)
+
+        group_by_expressions = any(
+            gbc.expression != Expression.column_reference(gbc.name) for gbc in group_by_columns
+        )
+        if is_bigquery(self.engine) and group_by_expressions:
```

Comment:
naming was a bit odd
```suggestion
        has_group_by_expressions = any(
            gbc.expression != Expression.column_reference(gbc.name) for gbc in group_by_columns
        )
        if is_bigquery(self.engine) and has_group_by_expressions:
```

Reply:
yeah I think I like your naming better.

`group_by_expression` was supposed to read as 'do we group by an expression', but agreed that that's not clear.